src/main.ts

```diff
@@ -27,93 +27,91 @@ import { PaginationInterceptor } from './interceptors/pagination.interceptor';
 import { NestExpressApplication } from '@nestjs/platform-express';
 import { NftQueueModule } from './queue.worker/nft.worker/queue/nft.queue.module';
-import configuration from "config/configuration";
 import { PluginService } from './common/plugins/plugin.service';
 import { TransactionCompletedModule } from './crons/transaction.processor/transaction.completed.module';
 import { SocketAdapter } from './websockets/socket-adapter';
 import { RabbitMqProcessorModule } from './rabbitmq.processor.module';
+import { ApiConfigModule } from './common/api-config/api.config.module';
 
 async function bootstrap() {
-  const conf = configuration();
-  if (conf.flags?.useTracing === true) {
-    require('dd-trace').init();
-  }
+  const apiConfigApp = await NestFactory.create(ApiConfigModule);
+  const apiConfigService = apiConfigApp.get<ApiConfigService>(ApiConfigService);
 
-  const publicApp = await NestFactory.create<NestExpressApplication>(
-    PublicAppModule,
-  );
-  publicApp.use(bodyParser.json({ limit: '1mb' }));
-  publicApp.use(requestIp.mw());
-  publicApp.enableCors();
-  publicApp.useLogger(publicApp.get(WINSTON_MODULE_NEST_PROVIDER));
-  publicApp.disable('etag');
-  publicApp.disable('x-powered-by');
-
-  const apiConfigService = publicApp.get<ApiConfigService>(ApiConfigService);
-  const cachingService = publicApp.get<CachingService>(CachingService);
-  const httpAdapterHostService = publicApp.get<HttpAdapterHost>(HttpAdapterHost);
-  const metricsService = publicApp.get<MetricsService>(MetricsService);
-  const protocolService = publicApp.get<ProtocolService>(ProtocolService);
-  const pluginService = publicApp.get<PluginService>(PluginService);
-
-  if (apiConfigService.getIsAuthActive()) {
-    publicApp.useGlobalGuards(new JwtAuthenticateGuard(apiConfigService));
+  if (apiConfigService.getUseTracingFlag() === true) {
+    require('dd-trace').init();
   }
 
-  const httpServer = httpAdapterHostService.httpAdapter.getHttpServer();
-  httpServer.keepAliveTimeout = apiConfigService.getServerTimeout();
-  httpServer.headersTimeout = apiConfigService.getHeadersTimeout(); //`keepAliveTimeout + server's expected response time`
-
-  const globalInterceptors: NestInterceptor[] = [];
-  globalInterceptors.push(new LoggingInterceptor(metricsService));
-
-  if (apiConfigService.getUseRequestCachingFlag()) {
-    globalInterceptors.push(
-      new CachingInterceptor(
-        cachingService,
-        httpAdapterHostService,
-        metricsService,
-        protocolService,
-      ),
+  if (apiConfigService.getIsPublicApiActive()) {
+    const publicApp = await NestFactory.create<NestExpressApplication>(PublicAppModule);
+    publicApp.use(bodyParser.json({ limit: '1mb' }));
+    publicApp.use(requestIp.mw());
+    publicApp.enableCors();
+    publicApp.useLogger(publicApp.get(WINSTON_MODULE_NEST_PROVIDER));
+    publicApp.disable('etag');
+    publicApp.disable('x-powered-by');
+
```

Comment:
move this closer to their usage and define them only if need it

Reply:
these are part of the default configuration, have no usage in main.ts

Reply:
fixed